clcache.py

```diff
@@ -781,15 +788,10 @@ def expandBasedirPlaceholder(path):
 
 def collapseBasedirToPlaceholder(path):
     baseDir = normalizeBaseDir(os.environ.get('CLCACHE_BASEDIR'))
-    if baseDir is None:
-        return path
-    else:
-        assert path == os.path.normcase(path)
-        assert baseDir == os.path.normcase(baseDir)
-        if path.startswith(baseDir):
-            return path.replace(baseDir, BASEDIR_REPLACEMENT, 1)
-        else:
-            return path
+    if baseDir:
+        replacement = re.compile(re.escape(baseDir), re.IGNORECASE)
+        return replacement.sub(BASEDIR_REPLACEMENT, path)
+    return path
```

Comment:
Indeed I applied this change as a quick test and didn't give it too much thought. With the proposed implementation the `assert path == os.path.normcase(path)` does not hold. I can add back the other assert `assert baseDir == os.path.normcase(baseDir)`